src/createServer.js

```diff
@@ -1,7 +1,356 @@
+    if (!category) {
+      return res.status(404).json({ message: 'Category not found' });
+    }
+
+    res.json(category);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error fetching category', error: error.message });
+  }
+});
+
+router.post('/categories', async (req, res) => {
+  try {
+    const { name, description } = req.body;
+
+    if (!name) {
+      return res.status(400).json({ message: 'Category name is required' });
+    }
+
+    const newCategory = await Category.create({
+      name,
+      description: description || null,
+    });
+
+    res.status(201).json(newCategory);
+  } catch (error) {
+    res
+      .status(400)
+      .json({ message: 'Error creating category', error: error.message });
+  }
+});
+
+router.patch('/categories/:id', async (req, res) => {
+  try {
+    const { name, description } = req.body;
+
+    const category = await Category.findByPk(req.params.id);
+
+    if (!category) {
+      return res.status(404).json({ message: 'Category not found' });
+    }
+
+    if (name !== undefined) {
+      if (!name || name.trim() === '') {
+        return res
+          .status(400)
+          .json({ message: 'Category name cannot be empty' });
+      }
+
+      category.name = name;
+    }
+
+    if (description !== undefined) {
+      category.description = description;
+    }
+
+    await category.save();
+
+    res.json(category);
+  } catch (error) {
+    res
+      .status(400)
+      .json({ message: 'Error updating category', error: error.message });
+  }
+});
+
+router.delete('/categories/:id', async (req, res) => {
+  try {
+    const category = await Category.findByPk(req.params.id);
+
+    if (!category) {
+      return res.status(404).json({ message: 'Category not found' });
+    }
+
+    await category.destroy();
+
+    res.sendStatus(204);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error deleting category', error: error.message });
+  }
+});
+
+router.get('/expenses', async (req, res) => {
+  const { userId, category, from, to, categories } = req.query;
+
+  try {
+    const whereClause = {};
+
+    if (userId) {
+      whereClause.userId = userId;
+    }
+
+    if (category) {
+      whereClause.category = category;
+    }
+
+    if (categories) {
+      const categoryArray = Array.isArray(categories)
+        ? categories
+        : categories.split(',').map((cat) => cat.trim());
+
+      whereClause.category = categoryArray;
+    }
+
+    if (from && to) {
+      const start = new Date(from);
+      const end = new Date(to);
+
+      if (isNaN(start.getTime()) || isNaN(end.getTime())) {
+        return res.status(400).json({ message: 'Invalid date format' });
+      }
+
+      whereClause.spentAt = {
+        [Op.between]: [start, end],
+      };
+    }
+
+    const expenses = await Expense.findAll({ where: whereClause });
+
+    res.json(expenses);
+  } catch (error) {
+    res.status(500).json({ message: 'Error fetching expenses' });
```

Comment:
Consider including the error message in the response to provide more context for debugging.